src/nvim/api/private/helpers.c

```diff
@@ -604,6 +605,9 @@ void modify_keymap(Buffer buffer, bool is_unmap, String mode, String lhs, String
     return;
   }
 
+  if (channel_id == LUA_INTERNAL_CALL && opts != NULL && opts->func.type == kObjectTypeLuaRef) {
```

Comment:
no need to check `channel_id` , only a lua caller can produce a `kObjectTypeLuaRef` (and if there was a way, it wouldn't be a problem either..)

Reply:
Ok I'll remove those.